scylla-gdb.py

```diff
@@ -6089,6 +6089,94 @@ def invoke(self, arg, for_tty):
+        parser.add_argument("-k", "--keyspace", action="store", type=str,
+                help="Print tablet metadata for tables belonging to the given keyspace")
+        parser.add_argument("-t", "--table", action="store", type=str,
+                help="Print tablet metadata for the given table only (match by name, use together with --keyspace for an exact match)")
+
+        try:
+            args = parser.parse_args(arg.split())
+        except SystemExit:
+            return
+
+        db = find_db()
+        token_metadata = seastar_lw_shared_ptr(db['_shared_token_metadata']['_shared']).get()
+        tablet_metadata = std_unique_ptr(token_metadata['_impl']).get()['_tablets']
+        topology = std_unique_ptr(token_metadata['_impl']).get()['_topology']
+        this_node = topology['_this_node'].dereference()
+
+        gdb.write(f'This node: host_id: {this_node["_host_id"]["id"]}, shard: {current_shard()}\n\n')
+
+        def format_replica_set(replica_set):
+            return ", ".join([f'{replica["host"]["id"]}#{replica["shard"]}' for replica in small_vector(replica_set)])
+
+        tables = {}
+        for table in for_each_table(db):
+            schema = seastar_lw_shared_ptr(table['_schema']).get().dereference()
+            r = schema['_raw']
+            tables[str(r['_id']['id'])] = (str(r['_ks_name'])[1:-1], str(r['_cf_name'])[1:-1])
+
+        for table_id, tablet_map in unordered_map(tablet_metadata['_tablets']):
+            table_id = str(table_id['id'])
+            keyspace_name, table_name = tables[table_id]
+
+            if args.keyspace and keyspace_name != args.keyspace:
+                continue
+
+            if args.table and table_name != args.table:
+                continue
+
+            tablet_map = seastar_lw_shared_ptr(tablet_map['_value']).get()
+            log2_tablets = int(tablet_map['_log2_tablets'])
+            tablet_count = 2 ** log2_tablets
+            resize_decision = std_variant(tablet_map['_resize_decision']['way']).get().type.name.replace('locator::resize_decision::', '')
+            resize_decision_seq_num = int(tablet_map['_resize_decision']['sequence_number'])
+
+            tablet_transitions = {int(tablet_id['id']): transition_info
+                                  for tablet_id, transition_info in unordered_map(tablet_map['_transitions'])}
+
+            gdb.write(f'table {keyspace_name}.{table_name}: id: {table_id}, tablets: {tablet_count}, resize decision: {resize_decision}#{resize_decision_seq_num}, transitions: {len(tablet_transitions)}\n')
+
+            for tablet_id, tablet_info in enumerate(chunked_vector(tablet_map['_tablets'])):
+                last_token = self._last_token_of_compaction_group(log2_tablets, tablet_id)
+                replicas = format_replica_set(tablet_info['replicas'])
+                gdb.write(f'  tablet#{tablet_id}: last token: {last_token}, replicas: [{replicas}]\n')
+                if tablet_id in tablet_transitions:
```

Comment:
now there's repair info in tablet info, do you want to dump it in this command or introduce a specialized command for it?

Reply:
I would prefer to do it in a follow-up, when the case arises. The command is already incomplete, I only dumped what I had in the core I was developing this with.